sys/kern/vfs_bio.c

```diff
@@ -4888,17 +4888,25 @@ vm_hold_free_pages(struct buf *bp, int newbsize)
 int
 vmapbuf(struct buf *bp, int mapbuf)
 {
+	void * __capability b_data;
 	vm_prot_t prot;
 	int pidx;
+#if __has_feature(capabilities) && !defined(__CHERI_PURE_CAPABILITY__)
+	b_data = cheri_capability_build_user_data(CHERI_CAP_USER_DATA_PERMS,
+	    (__cheri_addr vaddr_t)bp->b_data, bp->b_bufsize, 0);
```

Comment:
_cheri_capability_build_user_rwx will assert if the cap does not end up being representable

Reply:
This is likely a problem at least some times. Assuming it's possible I think I'd like to land this change and then get `struct buf` and `struct bio` updated to use capabilities.  I fear that's going to be quite a grind (which is why I punted in the first place...)

Reply:
Actually, I'd rather change vmapbuf() to just pass the user pointer in as a direct argument.  I believe it is rarely used and that will be a much smaller change than changing `b_data`'s type which should only ever hold kernel pointers.